src/sentry/api/endpoints/organization_missing_org_members.py

```diff
@@ -56,12 +59,31 @@ def _get_missing_members(self, organization: Organization) -> QuerySet[CommitAut
             .order_by("-commit_count")
         )
 
-    # TODO(cathy): check domain
+    def _get_email_domain(self, email: str) -> str:
+        return Address(addr_spec=email).domain
 
     def get(self, request: Request, organization) -> Response:
-        # TODO(cathy): search
         queryset = self._get_missing_members(organization)
 
+        # if a member has user_email=None, then they have yet to accept an invite
+        org_owners = organization.get_members_with_org_roles(
+            roles=[roles.get_top_dog().id]
+        ).exclude(user_email=None)
+
+        owner_emails = {self._get_email_domain(owner.user_email) for owner in org_owners}
+
+        if len(owner_emails) == 1:
+            queryset = queryset.filter(email__endswith=owner_emails.pop())
+
+        query = request.GET.get("query")
+        if query:
```

Comment:
this is how we get a search bar to work

Reply:
`tokenize_query` parses all of the query params and formats them nicely so you can iterate through them. you can have `&query=hello&otherquery=hola` in your url and it would spit them out like `{"query": "hello", "otherquery", "hola"}`